src/rules/arrowReturnShorthandRule.ts

```diff
@@ -42,59 +43,60 @@ export class Rule extends Lint.Rules.AbstractRule {
-        const text = this.getSourceFile().text;
-        const statement = expr.parent!;
-        const returnKeyword = Lint.childOfKind(statement, ts.SyntaxKind.ReturnKeyword)!;
-        const arrow = Lint.childOfKind(arrowFunction, ts.SyntaxKind.EqualsGreaterThanToken)!;
-        const openBrace = Lint.childOfKind(body, ts.SyntaxKind.OpenBraceToken)!;
-        const closeBrace = Lint.childOfKind(body, ts.SyntaxKind.CloseBraceToken)!;
-        const semicolon = Lint.childOfKind(statement, ts.SyntaxKind.SemicolonToken);
+function createFix(arrowFunction: ts.FunctionLikeDeclaration, body: ts.Block, expr: ts.Expression, text: string): Lint.Fix | undefined {
+    const statement = expr.parent!;
+    const returnKeyword = Lint.childOfKind(statement, ts.SyntaxKind.ReturnKeyword)!;
+    const arrow = Lint.childOfKind(arrowFunction, ts.SyntaxKind.EqualsGreaterThanToken)!;
+    const openBrace = Lint.childOfKind(body, ts.SyntaxKind.OpenBraceToken)!;
+    const closeBrace = Lint.childOfKind(body, ts.SyntaxKind.CloseBraceToken)!;
+    const semicolon = Lint.childOfKind(statement, ts.SyntaxKind.SemicolonToken);
 
-        const anyComments = hasComments(arrow) || hasComments(openBrace) || hasComments(statement) || hasComments(returnKeyword) ||
-            hasComments(expr) || (semicolon && hasComments(semicolon)) || hasComments(closeBrace);
-        return anyComments ? undefined : [
-            // Object literal must be wrapped in `()`
-            ...(expr.kind === ts.SyntaxKind.ObjectLiteralExpression ? [
-                this.appendText(expr.getStart(), "("),
-                this.appendText(expr.getEnd(), ")"),
-            ] : []),
-            // " {"
-            this.deleteFromTo(arrow.end, openBrace.end),
-            // "return "
-            this.deleteFromTo(statement.getStart(), expr.getStart()),
-            // " }" (may include semicolon)
-            this.deleteFromTo(expr.end, closeBrace.end),
-        ];
+    const anyComments = hasComments(arrow) || hasComments(openBrace) || hasComments(statement) || hasComments(returnKeyword) ||
+        hasComments(expr) || (semicolon && hasComments(semicolon)) || hasComments(closeBrace);
+    return anyComments ? undefined : [
+        // Object literal must be wrapped in `()`
+        ...(expr.kind === ts.SyntaxKind.ObjectLiteralExpression ? [
+            Lint.Replacement.appendText(expr.getStart(), "("),
+            Lint.Replacement.appendText(expr.getEnd(), ")"),
+        ] : []),
+        // " {"
+        Lint.Replacement.deleteFromTo(arrow.end, openBrace.end),
+        // "return "
+        Lint.Replacement.deleteFromTo(statement.getStart(), expr.getStart()),
+        // " }" (may include semicolon)
+        Lint.Replacement.deleteFromTo(expr.end, closeBrace.end),
+    ];
 
-        function hasComments(node: ts.Node): boolean {
-            return ts.getTrailingCommentRanges(text, node.getEnd()) !== undefined;
-        }
+    function hasComments(node: ts.Node): boolean {
```

Comment:
there is a function called `hasCommentAfterPosition` in `src/language/utils.ts` which does basically the same and will also handle leading trivia of the next token